mass-merge.py

```diff
@@ -0,0 +1,203 @@
+            error(f"ERROR on filename '{db}'.")
+            error("No manifest, but a manifest is required.")
+            sys.exit(-1)
+
+        ident_picklist = SignaturePicklist('ident')
+        ident_picklist.pickset = all_idents
+
+        check_idx = idx.select(ksize=args.ksize,
+                            moltype=moltype,
+                            picklist=ident_picklist)
+        found_idents.update(ident_picklist.found)
+        idx_list.append(idx)
+
+    # make sure that we get all the things.
+    if not all_idents.issubset(found_idents):
+        remaining = all_idents - found_idents
+        error(f"ERROR: {len(remaining)} identifiers from spreadsheet not found.")
+        example_missing = "\n".join(remaining)
+        error(f"Here are some examples: {example_missing}")
+        sys.exit(-1)
+
+    if args.check:
+        notify("Everything looks copacetic. Exiting as requested by `--check`")
+        sys.exit(0)
+
+    notify("Everything looks copacetic. Proceeding to merge!")
+
+    # go through, do merge, save.
+    with sourmash_args.SaveSignaturesToLocation(args.output) as save_sigs:
+        n=0
+        n_singletons = 0
+        for m, (merge_name, idents) in enumerate(merge_d.items()):
+            # keep count of singletons
+            if len(idents) == 1:
+                n_singletons+=1
+            if m % 100 == 0:
+                merge_percent = float(n)/len(found_idents) * 100
+                notify(f"...merging sigs for {merge_name} ({merge_percent:.1f}% of sigs merged)", end="\r")
+
+            # build a new picklist for idents to be merged
+            ident_picklist = SignaturePicklist('ident')
+            ident_picklist.pickset = set(idents)
+
+            # loop through each idx (db), select sigs, merge mh
+            first_sig = None
+            mh = None
+            merged_ss = None
+            for idx in idx_list:
+                this_idx = idx.select(ksize=args.ksize,
+                                moltype=moltype,
+                                picklist=ident_picklist)
+                for ss in this_idx.signatures():
+                    n += 1
+                    # first sig? initialize some things
+                    if first_sig is None:
+                        first_sig = ss
+                        mh = first_sig.minhash.copy_and_clear()
+
+                        # forcibly remove abundance?
+                        if args.flatten:
+                            mh.track_abundance = False
+
+                    try:
+                        sigobj_mh = ss.minhash
+                        if not args.flatten:
+                            _check_abundance_compatibility(first_sig, ss)
+                        else:
+                            sigobj_mh.track_abundance = False
+
+                        mh.merge(sigobj_mh)
+                    except (TypeError, ValueError) as exc:
+                        error("ERROR when merging signature '{}' ({}) from file {}",
```

Comment:
you have three placeholders, and only two values. Suggest using f-string.

Reply:
thanks! Hazards of swiping the `sig merge` code :)